src/functional/ffront/foast_passes/dead_closure_var_elimination.py

```diff
@@ -0,0 +1,50 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2022, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either vervisit_Constantsion 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+from typing import Any
+
+import functional.ffront.field_operator_ast as foast
+from eve import NodeTranslator, traits
+
+
+class DeadClosureVarElimination(NodeTranslator, traits.VisitorWithSymbolTableTrait):
+    _referenced_symbols: list[foast.Symbol]
+
+    @classmethod
+    def apply(cls, node: foast.FieldOperator):
+        return cls().visit(node)
+
+    def visit_Name(self, node: foast.Name, **kwargs: Any) -> foast.Name:
+        symtable = kwargs["symtable"]
+        if node.id in symtable:
```

Comment:
You need to check not only the id, but the actual declaration.
```suggestion
        if node.id in symtable and symtable[node.id] is closure_vars[node.id]:
```

Reply:
It's collecting all referenced symbols, not just the closure vars. The if is only to avoid dealing with undefined symbols.